### PR TITLE
extract jscs config into .jscsrc; resolves #632

### LINKED ISSUE #632: extract JSCS config into .jscsrc
JetBrains IDEs, and probably other text editors, support inline JSCS linting.  However, as long as the JSCS config is in the Gruntfile, they don't know what rules to use.

Cut the JSCS config from `Gruntfile.js` and paste it into a new `.jscsrc`.

### PR DESCRIPTION
**Pull request by boneskull**
- created `jscs` npm script target; `npm run-script jscs`
- removed JSCS config from `Gruntfile.js`; config now lives in `.jscsrc`
- added `jscs` as devDependency
- added `util/` to paths covered by JSCS
- fixed JSCS errors in `util/xbee.js`
  - a couple formatting problems
  - you can't `return` a `process.exit()` call
  - fixed what appeared to be a logic error when no port name specified; would exit too early
  - the above three issues were reported by JSCS

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,84 +8,325 @@
...
+      "email": "<danyshaanan@gmail.com>"
+    },
+    {
+      "name": "Rachel Hazes",
+      "email": "<>"
+    },
+    {
+      "name": "DeShawn Williams",
+      "email": "<deshawn.b.williams@gmail.com>"
+    }
```
Comment (rwaldron) [inline package.json]: Nope.

File: package.json [hunk 2]
```diff
@@ -8,84 +8,325 @@
...
+    "spark",
+    "spark core",
+    "spark-io",
+    "raspberrypi",
+    "raspberry pi",
+    "rpi",
+    "raspi-io",
+    "intel galileo",
+    "galileo",
+    "galileo-io"
```
Comment (rwaldron) [inline package.json]: Undo please

**Comment by rwaldron** [comment]
> - created jscs npm script target; npm run-script jscs
> - added jscs as devDependency

But Johnny-Five uses grunt :\

File: Gruntfile.js [hunk 3]
```diff
@@ -108,41 +108,9 @@ module.exports = function(grunt) {
...
     jscs: {
-      files: {
-        src: [
-          "Gruntfile.js",
-          "lib/**/!(johnny-five)*.js",
-          "test/**/*.js",
-          "eg/**/*.js",
-        ]
-      },
+      src: "**/*.js",
```
Comment (rwaldron) [inline Gruntfile.js]: I have a local directory called `wip` that's full of junk files, experiments, copy/paste crap, etc. I don't want those files to be reviewed by jscs, as they will most certainly cause false failures locally.

File: package.json (hunk 2 above)
Comment (boneskull) [inline package.json]: oops.  overactive IDE

File: Gruntfile.js (hunk 3 above)
Comment (boneskull) [inline Gruntfile.js]: ok, I'll make this explicit.  ok to leave `util/`?

**Comment by boneskull** [comment]
> But Johnny-Five uses grunt :\

I understand.  I can remove the devDep.  If you want to leave the npm script though, I could change it to execute "grunt jscs" instead.  

I do stuff like this in my projects just to make things more accessible.  Otherwise, I can remove it.

**Comment by boneskull** [comment]
@rwaldron Pushed requested changes

**Comment by boneskull** [comment]
@rwaldron Let me know if it's ok and I'll squash

File: Gruntfile.js [hunk 4]
```diff
@@ -108,41 +108,15 @@ module.exports = function(grunt) {
...
-          "test/**/*.js",
-          "eg/**/*.js",
-        ]
-      },
+      src: [
+        "Gruntfile.js",
+        "lib/**/!(johnny-five)*.js",
+        "test/**/*.js",
+        "eg/**/*.js",
+        "util/**/*.js"
```
Comment (rwaldron) [inline Gruntfile.js]: :+1:

**Comment by rwaldron** [comment]
@boneskull perfect, squash away!

**Comment by boneskull** [comment]
@rwaldron done

**Comment by divanvisagie** [comment]
:+1:

**Pull request merged**